public/lesswrong-editor/src/ckeditor5-footnote/src/footnoteEditing/converters.js

```diff
@@ -0,0 +1,395 @@
+// @ts-check (uses JSDoc types for type checking)
+
+import { Editor } from '@ckeditor/ckeditor5-core';
+import { toWidget, toWidgetEditable } from '@ckeditor/ckeditor5-widget/src/utils';
+import ContainerElement from '@ckeditor/ckeditor5-engine/src/view/containerelement';
+import { DowncastConversionApi } from '@ckeditor/ckeditor5-engine/src/conversion/downcastdispatcher';
+import ModelElement from '@ckeditor/ckeditor5-engine/src/model/element';
+import ViewElement from '@ckeditor/ckeditor5-engine/src/view/element';
+import RootElement from '@ckeditor/ckeditor5-engine/src/model/rootelement';
+import TextProxy from '@ckeditor/ckeditor5-engine/src/model/textproxy';
+import Node from '@ckeditor/ckeditor5-engine/src/model/node';
+import { modelQueryElementsAll, viewQueryElement, viewQueryText } from '../utils';
+import { ATTRIBUTES, CLASSES, ELEMENTS } from '../constants';
+
+/**
+ * Defines methods for converting between model, data view, and editing view representations of each element type.
+ * @param {Editor} editor
+ * @param {RootElement} rootElement
+ * @returns {void}
+ * */
+export const defineConverters = (editor, rootElement) => {
+	const conversion = editor.conversion;
+
+	/***********************************Footnote Section Conversion************************************/
+	// ((data) view → model)
```

Comment:
CKEditor is agnostic about what an HTML element in the view layer should map to in the model layer until you tell it otherwise, and vice versa. This upcast converter in particular fires when HTML is loaded from e.g. the database or local storage, and CKEditor requires a mapping if I want any of those HTML elements to be converted into rich model elements that the functionality of this plugin relies on. 

You could in principle rely solely on the data attributes that get passed from view to model for most functionality without upcasting to a named element, but it's a bit more verbose/clunky, it's less semantically clear, and there are some APIs that you won't have easy access to (like listening to events that fire on elements with a given name).